### DISCUSSION TITLE
🦋 🐛 🐝 🐞 🐜 🕷 🕸 🦂 🦗 🦟

### DISCUSSION THREAD

**Discussion by hayashi-ay** [katataku/minishell#69] [General]
🦋 🐛 🐝 🐞 🐜 🕷 🕸 🦂 🦗 🦟

**Comment by hayashi-ay** [comment]
Ctrl+Dを入力したときに、exitが改行されて表示されてしまう。
仕様にしても良さそうかな。

bash
```bash
bash-3.2$ false
bash-3.2$ exit
c1r31s4% 
```

./minishell
```bash
]$ ./minishell 
> false
> 
exit
```

**Comment by hayashi-ay** [comment]
パイプの前にコマンドがないときにエラーにしていない

bash

```bash
$ | pwd
bash: syntax error near unexpected token `|'
```

./minishell

```bash
| pwd
/Users/hayashi-ay/Projects/minishell
```

**Comment by hayashi-ay** [comment]
スペースだけを入力してenterを押すとセグフォ

```bash
./minishell 
>  
Segmentation fault: 11
```

**Comment by hayashi-ay** [comment]
<img width="575" alt="スクリーンショット 2022-03-22 23 01 14" src="https://user-images.githubusercontent.com/39038230/159499102-e65b04a4-e78b-4720-9763-58720ed2fae5.png">

last exit statusに値が入った後にコマンドが実行されるとセグフォになるっぽい。

**Reply by hayashi-ay** [reply]
このif文をコメントアウトしたら動いた
https://github.com/katataku/minishell/blob/main/srcs/lexer/lexer.c#L124

**Reply by hayashi-ay** [reply]
lexer単体としては正常に動いているが、結合したときにlexerがNULLを返す場合を後続で考慮できていない感じっぽい。

**Reply by hayashi-ay** [reply]
main側でハンドリングするか、lexer側でハンドリングするか

**Comment by katataku** [comment]
127が帰ってこない

```bash
$ ./minishell
> hogehoge
hogehoge: command not found
> echo $?
0
```

**Comment by hayashi-ay** [comment]
builtin単体のときにヒアドキュメントが実行できていない。そもそもheredocはどこでやる？

<img width="456" alt="スクリーンショット 2022-03-24 14 32 09" src="https://user-images.githubusercontent.com/39038230/159849261-37453e0a-be05-45c9-8eb1-b67b810fa9dc.png">

**Comment by hayashi-ay** [comment]
区切り文字、スペースだけでなくてタブなどもありそう。
```
echo -n "$IFS" | od -b
```

**Comment by katataku** [comment]
```bash
c1r19s8% ./minishell 
> ^C
> ^C
> echo $?
0
> 
exit
c1r19s8% bash 
bash-3.2$ 
bash-3.2$ 
bash-3.2$ echo $?
1
bash-3.2$ 
```

**Comment by katataku** [comment]
```bash
bash-3.2$ echo $?
130
bash-3.2$ exit
c1r19s8% ./minishell 
> cat
^C> echo $?
0
```

**Comment by katataku** [comment]
SIGQUITはmacだと効かないかも？あるいはJISキーボードだと効かないかも？
現状無視しているので再確認が必要
